packages/x-data-grid/src/hooks/features/density/densitySelector.ts

```diff
@@ -4,16 +4,29 @@ import { GridDensity } from '../../../models/gridDensity';
 
 export const COMPACT_DENSITY_FACTOR = 0.7;
 export const COMFORTABLE_DENSITY_FACTOR = 1.3;
+export const COMPACT_DENSITY_FACTOR_HEADER = 0.8;
+export const COMFORTABLE_DENSITY_FACTOR_HEADER = 1.3;
```

Comment:
We may want to lower this at some point, as `comfortable` column headers are quite tall.